inject.js

```diff
@@ -444,10 +477,35 @@
         try { var childDocument = frame.contentDocument } catch (e) { return }
         initializeWhenReady(childDocument);
       });
+
+      //look for video in shadowRoot for apple tv
+      var apple_tv = document.querySelector('apple-tv-plus-player')
```

Comment:
Instead of creating a dedicated observer, can we refactor and merge the logic with the existing MutationObserver? On that note a couple of related questions..

- Why do we need to detect aria-hidden? 
- Same as above, can we refactor to check for shadowRoot and traverse without special casing Apple TV?

Reply:
> Instead of creating a dedicated observer, can we refactor and merge the logic with the existing MutationObserver? 

yes. 

> Why do we need to detect aria-hidden?

The video tag is not in any part of the DOM (shadow or regular) until the the aria-hidden attribute value changes to false.

> can we refactor to check for shadowRoot and traverse without special casing Apple TV?

I set a breakpoint on initializeNow, and found out that the shadow roots were empty. I assume that this is an optimization done by apple to get the page loaded quickly and then lazy initialize the player later. it might be possible to set a mutationObserver.
However, mutationObservers don't cross into the shadowRoot. see: https://github.com/w3c/webcomponents/issues/390 and https://github.com/w3c/webcomponents/issues/204. In the ally.js [project,](https://github.com/medialize/ally.js/blob/master/src/observe/shadow-mutations.js) their implementation is to have a mutationObserver look for shadowRoot creation and then attach another recursive mutationObserver again to the shadowRoot. Since there are like 18 shadowRoots that are densely nested on the apple tv plus website, intuitively, creating a lot of mutationObservers will negatively impact performance.
Also, I looked at slotChangeEvents but it didn't work because of the shadowDOM. I also looked at this https://www.polymer-project.org/blog/shadydom and this https://github.com/Polymer/polymer/blob/master/lib/utils/flattened-nodes-observer.js